### PR TITLE
Feature event status boxes

### PR DESCRIPTION
**Pull request by nesteves**
Implements the functionality requested in issue #392 

The event's page now shows one set of status boxes for each department:

rr_event_status

Summarizing the changes to the codebase:

* a bit of housekeeping on the files I worked with: deleted unused helper modules; tried to split the event feature specs by view or major piece of functionality; refactored the event and person model specs; added a spec to the assignment model
* refactored the methods that fetch an event's responses/people and upon which this feature depends so they could be chained as a normal relation. I chose not to unit test these as they call on other model scopes (_Availability_ and _Person_) that are already covered
* implemented the necessary changes to the view and helper

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -51,51 +51,62 @@ def to_s
...
   end
 
   def responding_people
-    self.responses.map { |a| a.person }
+    Person.active.joins(:availabilities).
+      merge(Availability.containing(start_time..end_time))
   end
 
   def eligible_people
-    Person.active.where(department: self.departments).where("title_order <= ?", Person::TITLE_ORDER[min_title])
```
Comment (kgf) [inline app/models/event.rb:96]: This change is incorrect, it should be `<=`.
1 is the most important title. 100 would be a less important title.
 Ideally we should have a test on this, as we've had this regression before.

**Comment by kgf** [comment]
@nesteves Thanks for your help on this ! One small needed change, and a request for a test if you could...

Thanks !

File: app/models/event.rb (hunk 1 above)
Comment (nesteves) [reply app/models/event.rb:96]: Oh, so a person ranked higher than another should have a `TITLE_ORDER` that is *less than* the other person's.

Thanks for the heads up!

**Comment by kgf** [comment]
@nesteves Thanks for the quick turn around. Sadly, the tests failed.

**Comment by kgf** [comment]
Fixes #392 
Thanks for your help @nesteves !

**Pull request merged**